internal/services/netapp/netapp_volume_group_sap_hana_resource.go

```diff
@@ -0,0 +1,641 @@
+					if err != nil {
+						return err
+					}
+
+					// Authorizing
+					if err = replicationClient.VolumesAuthorizeReplicationThenPoll(ctx, pointer.From(primaryId), volumesreplication.AuthorizeRequest{
+						RemoteVolumeResourceId: utils.String(secondaryId.ID()),
+					},
+					); err != nil {
+						return fmt.Errorf("cannot authorize volume replication: %v", err)
+					}
+
+					// Wait for volume replication authorization to complete
+					log.Printf("[DEBUG] Waiting for replication authorization on %s to complete", id)
+					if err := waitForReplAuthorization(ctx, replicationClient, pointer.From(primaryId)); err != nil {
+						return err
+					}
+				}
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppVolumeGroupSapHanaResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			volumeClient := metadata.Client.NetApp.VolumeClient
+
+			id, err := volumegroups.ParseVolumeGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state NetAppVolumeGroupSapHanaModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Updating %s", id)
+
+			if metadata.ResourceData.HasChange("volume") {
+
+				// Iterating over each volume and performing individual patch
+				for i := 0; i < metadata.ResourceData.Get("volume.#").(int); i++ {
+
+					// Checking if individual volume has a change
+					volumeItem := fmt.Sprintf("volume.%v", i)
+
+					capacityPoolId, err := capacitypools.ParseCapacityPoolID(metadata.ResourceData.Get(fmt.Sprintf("%v.capacity_pool_id", volumeItem)).(string))
+					if err != nil {
+						return err
+					}
+
+					if metadata.ResourceData.HasChange(volumeItem) {
+
+						volumeId := volumes.NewVolumeID(id.SubscriptionId,
+							id.ResourceGroupName,
+							id.NetAppAccountName,
+							capacityPoolId.CapacityPoolName,
+							metadata.ResourceData.Get(fmt.Sprintf("%v.name", volumeItem)).(string))
+
+						update := volumes.VolumePatch{
+							Properties: &volumes.VolumePatchProperties{},
+						}
+
+						if metadata.ResourceData.HasChange(fmt.Sprintf("%v.storage_quota_in_gb", volumeItem)) {
+							storageQuotaInBytes := int64(metadata.ResourceData.Get(fmt.Sprintf("%v.storage_quota_in_gb", volumeItem)).(int) * 1073741824)
+							update.Properties.UsageThreshold = utils.Int64(storageQuotaInBytes)
+						}
+
+						if metadata.ResourceData.HasChange(fmt.Sprintf("%v.export_policy_rule", volumeItem)) {
+							exportPolicyRuleRaw := metadata.ResourceData.Get(fmt.Sprintf("%v.export_policy_rule", volumeItem)).([]interface{})
+
+							// Validating export policy rules
+							volumeProtocolRaw := (metadata.ResourceData.Get(fmt.Sprintf("%v.protocols", volumeItem)).([]interface{}))[0]
```

Comment:
This check might fail when multiple protocols are supported as someone could pass in `["randomProtocol", "nsfv3"]` and then set `rule.Nfsv41` to `true` and it wouldn't fail like it is supposed to. We should pass in the whole protocols slice to `ValidateNetAppVolumeGroupExportPolicyRuleSAPHanna`.

Reply:
Schema does not allow the multiple protocol, only one, should we write a validation for that anyways?